include/revng-c/Support/DecompilationHelpers.h

```diff
@@ -41,19 +41,19 @@ inline bool mayReadMemory(const llvm::Instruction &I) {
 
 /// Check if \a ModelType can be assigned to an llvm::Value of type \a LLVMType
 /// during a memory operations (load, store and the like).
-inline bool areMemOpCompatible(const model::QualifiedType &ModelType,
+inline bool areMemOpCompatible(const model::AnyType auto &ModelType,
```

Comment:
What's the advantage of having a template argument as opposed to use the base class?

Reply:
Can't you use `model::Type`? Is this supposed to catch `model::TypeDefinition` as well? O_o

Reply:
I definitely did at some point. Can't say if it still does from the top of my head though.

Reply:
Done.